include/BSMPT/minimum_tracer/minimum_tracer.h

```diff
@@ -0,0 +1,830 @@
+  int WhichMinimizer;
+  bool UseMultithreading;
+
+protected:
+  /**
+   * @brief modelPointer for the used parameter point
+   */
+  std::shared_ptr<Class_Potential_Origin> modelPointer;
+
+public:
+  /**
+   * @brief Threshold for the acceptable gradient
+   *
+   */
+  double GradientThreshold = 1e-3;
+
+  /**
+   * @brief Minimum found in IsThereEWSymmetryRestoration()
+   *
+   */
+  std::vector<double> HighTemperatureVEV;
+
+  /**
+   * @brief Vector to store minima that appeared from VEV splittings
+   *
+   */
+  std::vector<Minimum> SavedMinimaFromVEVSplitting;
+
+  /**
+   * @brief default constructor
+   */
+  MinimumTracer();
+
+  /**
+   * @brief constructor
+   * @param pointer_in this->modelPointer for used parameter point
+   * @param WhichMinimizer_in which minimizers are used
+   * @param UseMultithreading_in whether or not multithreading is used
+   */
+  MinimumTracer(std::shared_ptr<Class_Potential_Origin> &pointer_in,
+                const int &WhichMinimizer_in,
+                const bool &UseMultithreading_in);
+
+  /**
+   * @brief Calculates flat field directions
+   */
+  void FindFlatDirections();
+
+  /**
+   * @brief Convert point into minimal non-flat space, reduces dimension in case
+   * of flat directions, point has to have VEV dimension
+   */
+  void ConvertToNonFlatDirections(std::vector<double> &point);
+
+  /**
+   * @brief Calculates the list of symmetries that leave V unchanged
+   *
+   */
+  void FindDiscreteSymmetries();
+
+  /**
+   * @brief ConvertToVEVDim converts point from full to reduced (VEV) dimension
+   * @param point point in full field dimension
+   * @return point in reduced VEV dimension
+   */
+  std::vector<double> ConvertToVEVDim(const std::vector<double> &point);
+
+  /**
+   * @brief get global minimum of effective potential
+   * @param Temp temperature
+   * @param check storage for minimization debugging options
+   * @param start start value for CMA-ES minimization
+   * @return global minimum at temperature Temp
+   */
+  std::vector<double> GetGlobalMinimum(const double &Temp,
+                                       std::vector<double> &check,
+                                       const std::vector<double> &start);
+
+  /**
+   * @brief get global minimum of effective potential
+   * @param Temp temperature
+   * @param start start value for CMA-ES minimization
+   * @return global minimum at temperature Temp
+   */
+  std::vector<double> GetGlobalMinimum(const double &Temp,
+                                       const std::vector<double> &start);
+
+  /**
+   * @brief get global minimum of effective potential
+   * @param Temp temperature
+   * @return global minimum at temperature Temp
+   */
+  std::vector<double> GetGlobalMinimum(const double &Temp);
+
+  /**
+   * @brief IsGlobMin checks whether current minimum is the global minimum
+   * @param min Minimum to check, sets is_glob_min to true if min is global
+   * minimum, to false if min is local minimum
+   */
```

Comment:
Per default the variable is also false. Do you need to differentiate between the two false states or does this not matter?

Reply:
Doesn't matter, I drop this part of the description.